### DISCUSSION TITLE
How do I automatically fix warnings

### DISCUSSION THREAD

**Discussion by kjkoeller** [remarkjs/remark#1091] [Q&A] [answered]
I am brand new to this remark-lint and npm stuff so bare with me here.

Is there a way for remark to automatically update the `.md` files to fix the warnings that it brings up for each `.md` file? I have run this code in git bash `npx remark doc/ --use remark-preset-lint-consistent --use remark-preset-lint-recommended` and it brings up warnings and such. Then I tried the `npm run check` and `npm run format` but neither do anything. The files say they have been updated but when I paste the file contents onto my GitHub page, it says there are no changes to require a commit.

**Comment by Murderlon** [comment, marked as answer]
Hi, there is an example for it:
https://github.com/remarkjs/remark/tree/main/packages/remark-cli#example-checking-and-formatting-markdown-on-the-cli

In short: add the `--output` flag to your command.

**Reply by kjkoeller** [reply]
I assume to the first command? Also when I try any of them with -o at the end, it still shows the warnings and nothing has changed

**Reply by kjkoeller** [reply]
I also tried a single file with a manual --output at the end of the first command I typed and still there were no changes even though 26 warnings came up.

**Reply by ChristianMurphy** [reply]
Not all lint rules are fixable right now.
See the discussion at https://github.com/remarkjs/remark-lint/issues/82 for the long running and ongoing discussion to improve lint fixes.

**Reply by Murderlon** [reply]
In the meantime, you could use [`remark-language-server`](https://github.com/remarkjs/remark-language-server) or [`vscode-remark`](https://github.com/remarkjs/vscode-remark/) to have a better experience when dealing with non-fixable warnings.

**Reply by kjkoeller** [reply]
Thanks for the help!
